src/components/Modals/SendAssetModal.js

```diff
@@ -5,46 +5,75 @@ import { useForm } from 'react-hook-form';
 
 // COMPONENTS
 import { useDispatch } from 'react-redux';
+import { BN } from '@polkadot/util';
 import ModalRoot from './ModalRoot';
 import { TextInput } from '../InputComponents';
 import Button from '../Button/Button';
 
 // STYLES
 import styles from './styles.module.scss';
 import { parseAssets, isValidSubstrateAddress } from '../../utils/walletHelpers';
-import { walletActions } from '../../redux/actions';
+import { congressActions, senateActions, walletActions } from '../../redux/actions';
 import InputSearch from '../InputComponents/InputSearchAddressName';
+import Validator from '../../utils/validator';
+import useCongressExecutionBlock from '../../hooks/useCongressExecutionBlock';
 
 // TODO add validation
-function SendAssetModal({ closeModal, assetData }) {
+function SendAssetModal({
+  closeModal, assetData, isRemarkNeeded, isCongress,
+}) {
   const dispatch = useDispatch();
-
   const {
     handleSubmit,
     formState: { errors },
     register,
     setValue,
+    watch,
   } = useForm({
     mode: 'all',
+    defaultValues: {
+      votingDays: '7',
+    },
   });
+  const votingDays = watch('votingDays');
+  const executionBlock = useCongressExecutionBlock(votingDays);
 
   const transfer = (values) => {
-    dispatch(walletActions.sendAssetsTransfer.call({
-      recipient: values.recipient,
-      amount: parseAssets(values.amount, assetData.metadata.decimals),
-      assetData,
-    }));
+    const amount = parseAssets(values.amount, assetData.metadata.decimals);
+    if (!isRemarkNeeded) {
+      dispatch(walletActions.sendAssetsTransfer.call({
+        recipient: values.recipient,
+        amount,
+        assetData,
+      }));
+    } else {
+      const title = `Spend ${assetData.metadata.symbol}`;
+      const data = {
+        transferToAddress: values.recipient,
+        transferAmount: amount,
+        assetData,
+        remarkInfo: values.description || `${isCongress ? 'Congress' : 'Senate'} ${title}`,
+      };
+      if (isCongress) {
+        dispatch(congressActions.congressSendAssets.call({ ...data, executionBlock }));
+      } else {
+        dispatch(senateActions.senateSendAssets.call({ ...data, isCongress }));
+      }
+    }
     closeModal();
   };
-
+  const { balance } = assetData.balance;
   return (
     <form className={styles.getCitizenshipModal} onSubmit={handleSubmit(transfer)}>
       <div className={styles.h3}>
         Send
+        {' '}
         {assetData.metadata.symbol}
       </div>
       <div className={styles.description}>
-        You are going to send tokens from your wallet
+        {!isRemarkNeeded
```

Comment:
OOF

We ll need to refactor how modal decides whether its sending your tokens or govt tokens